### ISSUE TITLE
Vault crashes when attempting to generate-root while Vault is sealed

### ISSUE THREAD

**Issue by thomasmitchell** [hashicorp/vault#9310]
**Describe the bug**
If you attempt to initialize a new generate root token session but the Vault is sealed, the Vault server crashes.

**To Reproduce**
1. Start a `vault server -dev`
2. Seal the vault server
3. run `vault operator generate-root -generate-otp`
2. `vault operator generate-root -init -otp 'yourotp'`
3. See that the Vault has crashed

**Expected behavior**
I expected a returned error code telling me that the Vault is sealed.
**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.4.2
* Vault CLI Version (retrieve with `vault version`): 1.4.2
* Server Operating System/Architecture: linux/amd64 under WSL2 on Windows

Vault server configuration file(s):

`vault server -dev`

**Additional context**

```
nil pointer dereference
goroutine 374 [running]:
net/http.(*conn).serve.func1(0xc000a228c0)
        /usr/local/go/src/net/http/server.go:1767 +0x139
panic(0x30582c0, 0x5f6dcc0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
sync.(*RWMutex).RLock(...)
        /usr/local/go/src/sync/rwmutex.go:48
github.com/hashicorp/vault/vault.(*AuditBroker).LogRequest(0x0, 0x3e11320, 0xc0007efe30, 0xc00028b5e0, 0xc000944150, 0x0, 0x0)
        /go/src/github.com/hashicorp/vault/vault/audit_broker.go:92 +0x117
github.com/hashicorp/vault/vault.(*basicAuditor).AuditRequest(0xc0005f0340, 0x3e11320, 0xc0007efe30, 0xc00028b5e0, 0xc000a2a640, 0x0)
        /go/src/github.com/hashicorp/vault/vault/audit.go:519 +0x5f
github.com/hashicorp/vault/http.handleAuditNonLogical.func1(0x3defae0, 0xc0009d8540, 0xc000967000)
        /go/src/github.com/hashicorp/vault/http/handler.go:236 +0x2d0
net/http.HandlerFunc.ServeHTTP(0xc000b8b100, 0x3defae0, 0xc0009d8540, 0xc000967000)
        /usr/local/go/src/net/http/server.go:2007 +0x44
github.com/hashicorp/vault/http.handleRequestForwarding.func1(0x3defae0, 0xc0009d8540, 0xc000967000)
        /go/src/github.com/hashicorp/vault/http/handler.go:672 +0x2ac
net/http.HandlerFunc.ServeHTTP(0xc000b8b120, 0x3defae0, 0xc0009d8540, 0xc000967000)
        /usr/local/go/src/net/http/server.go:2007 +0x44
net/http.(*ServeMux).ServeHTTP(0xc0008f3480, 0x3defae0, 0xc0009d8540, 0xc000967000)
        /usr/local/go/src/net/http/server.go:2387 +0x1bd
github.com/hashicorp/vault/http.wrapHelpHandler.func1(0x3defae0, 0xc0009d8540, 0xc000967000)
        /go/src/github.com/hashicorp/vault/http/help.go:24 +0x155
net/http.HandlerFunc.ServeHTTP(0xc000b8b980, 0x3defae0, 0xc0009d8540, 0xc000967000)
        /usr/local/go/src/net/http/server.go:2007 +0x44
github.com/hashicorp/vault/http.wrapCORSHandler.func1(0x3defae0, 0xc0009d8540, 0xc000967000)
        /go/src/github.com/hashicorp/vault/http/cors.go:29 +0x93c
net/http.HandlerFunc.ServeHTTP(0xc000b8b9a0, 0x3defae0, 0xc0009d8540, 0xc000967000)
        /usr/local/go/src/net/http/server.go:2007 +0x44
github.com/hashicorp/vault/http.wrapGenericHandler.func1(0x3defae0, 0xc0009d8540, 0xc000966e00)
        /go/src/github.com/hashicorp/vault/http/handler.go:293 +0x33c
net/http.HandlerFunc.ServeHTTP(0xc000a377d0, 0x3defae0, 0xc0009d8540, 0xc000966e00)
        /usr/local/go/src/net/http/server.go:2007 +0x44
github.com/hashicorp/vault/vendor/github.com/hashicorp/go-cleanhttp.PrintablePathCheckHandler.func1(0x3defae0, 0xc0009d8540, 0xc000966e00)
        /go/src/github.com/hashicorp/vault/vendor/github.com/hashicorp/go-cleanhttp/handlers.go:42 +0xba
net/http.HandlerFunc.ServeHTTP(0xc000b8b9c0, 0x3defae0, 0xc0009d8540, 0xc000966e00)
        /usr/local/go/src/net/http/server.go:2007 +0x44
net/http.serverHandler.ServeHTTP(0xc0009d82a0, 0x3defae0, 0xc0009d8540, 0xc000966e00)
        /usr/local/go/src/net/http/server.go:2802 +0xa4
net/http.(*conn).serve(0xc000a228c0, 0x3e11260, 0xc000421bc0)
        /usr/local/go/src/net/http/server.go:1890 +0x875
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2928 +0x384
```

**Comment by aphorise** [comment]
@thomasmitchell this is already fixed and I think it was released by 1.4.7 at the time. Maybe confirm & close? 👍

**Comment by thomasmitchell** [comment]
can confirm it was fixed. 👍
